app/server/appsmith-server/src/main/java/com/appsmith/server/domains/Application.java

```diff
@@ -49,6 +50,25 @@
 
     String icon;
 
+    public Application(String name,
```

Comment:
Suggested signature for a copy constructor:
```suggestion
    public Application(Application originalApp) 
```
Inside this function, you then decide what values you want to copy and which you don't want to copy. Makes it cleaner to invoke.